src/page/upload/cf_upload.rs

```diff
@@ -57,16 +63,33 @@ mod cf_impl {
     pub mod server_func {
         use candid::Principal;
         use gob_cloudflare::{
-            api::stream_videos::{CreateDownloads, DirectUpload, VideoDetails},
+            api::stream_videos::{CreateDownloads, DirectUpload, EditVideoDetails, VideoDetails},
             CloudflareAuth,
         };
-        use leptos::{expect_context, ServerFnError};
+        use leptos::{expect_context, server_fn::ServerFn, ServerFnError};
 
         use crate::consts::CF_WATERMARK_UID;
 
         use super::UploadInfo;
         use std::time::Duration;
 
+        pub async fn edit_video_meta_impl(
+            uid: String,
+            post_id: u64,
+        ) -> Result<String, ServerFnError> {
+            let cf_api: CloudflareAuth = expect_context();
+            // get the existing meta from video
+            let req = VideoDetails::new(uid.clone());
+            let res = cf_api.send_auth(req).await?;
+            let meta = res.meta;
+
+            // add new meta to the video
+            let req = EditVideoDetails::new(uid, meta).add_meta("post_id", post_id.to_string());
```

Comment:
This is insecure.
We should perform a proper lookup in user's canister and only update if the uid matches for the given post id.